pkg/controller/lifecycle/actuator.go

```diff
@@ -179,6 +181,8 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extension
 		oidcShootAccessSecret.Secret.Name,
 		generatedSecrets[constants.WebhookTLSSecretName].Name,
 		k8sVersion,
+		// TODO(rfranzke): Delete this after August 2024.
+		a.client.Get(ctx, client.ObjectKey{Name: "prometheus-shoot", Namespace: ex.Namespace}, &appsv1.StatefulSet{}) == nil,
```

Comment:
Shouldn't we return the error if it is different from "not found"? 🤔

Reply:
There is not much value in doing so - we can simply use the "old contract" in that case. This contract is still valid and all this is just to ensure backwards-compatibility. After a few releases, only the "new contract" will be used, and then this code disappears again. Hence, I implemented the "minimal solution" to bridge this time gap.

Reply:
Ok, fair enough.